Assets/Scripts/GameManager.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.SceneManagement;
+
+public class GameManager : MonoBehaviour
+{
+
+    public GameObject Player;
+    public List<GameObject> CheckPoints;
+
+    private Vector3 _playerRespawnPosition;
+    private Quaternion _playerRespawnRotation;
+    private int _checkPointCount;
+    private PlayerController _playerController;
+
+    void Start()
+    {
+        _playerController = Player.GetComponent<PlayerController>();
+        _playerRespawnPosition = Player.transform.position;
+        _playerRespawnRotation = Player.transform.rotation;
+        _checkPointCount = 0;
+    }
+
+    void Update()
+    {
+        // When the player has reached a checkpoint, set the respawn point to it
+        if (CheckPoints.Count > 0 && _checkPointCount < CheckPoints.Count)
+        {   
```

Comment:
Should not use `Update` to check for new checkpoints. Instead update the checkpoint when player walks into checkpoint.